app/views/courses/CourseDetailsView.coffee

```diff
@@ -25,125 +25,64 @@ module.exports = class CourseDetailsView extends RootView
-      @onCourseSync()
-    else
-      @supermodel.loadModel @course
-
-  getRenderData: ->
-    context = super()
-    context.campaign = @campaign
-    context.course = @course if @course?.loaded
-    context.courseInstance = @courseInstance if @courseInstance?.loaded
-    context.courseInstances = @courseInstances?.models ? []
-    context.levelConceptMap = @levelConceptMap ? {}
-    context.noCourseInstance = @noCourseInstance
-    context.noCourseInstanceSelected = @noCourseInstanceSelected
-    context.userLevelStateMap = @userLevelStateMap ? {}
-    context.promptForSchool = @courseComplete and not me.isAnonymous() and not me.get('schoolName') and not storage.load('no-school')
-    context
-
-  afterRender: ->
-    super()
-    if @supermodel.finished() and @courseComplete and me.isAnonymous() and @options.justBeatLevel
-      # TODO: Make an intermediate modal that tells them they've finished HoC and has some snazzy stuff for convincing players to sign up instead of just throwing up the bare CreateAccountModal
-      CreateAccountModal = require 'views/core/CreateAccountModal'
-      @openModalView new CreateAccountModal showSignupRationale: true
-
-  onCourseSync: ->
-    return if @destroyed
-    # console.log 'onCourseSync'
-    if me.isAnonymous() and (not me.get('hourOfCode') and not @course.get('hourOfCode'))
-      @noCourseInstance = true
-      @render()
-      return
-    return if @campaign?
-    campaignID = @course.get('campaignID')
-    @campaign = @supermodel.getModel(Campaign, campaignID) or new Campaign _id: campaignID
-    @listenTo @campaign, 'sync', @onCampaignSync
-    if @campaign.loaded
-      @onCampaignSync()
-    else
-      @supermodel.loadModel @campaign
-    @render()
+    @levels = new Levels()
+    @courseInstances = new CourseInstances()
+
+    @supermodel.trackRequest @ownedClassrooms.fetchMine({data: {project: '_id'}})
+    @supermodel.trackRequest(@courses.fetch().then(=>
+      @course = @courses.get(@courseID)
+    ))
+    sessionsLoaded = @supermodel.trackRequest(@levelSessions.fetchForCourseInstance(@courseInstanceID, {cache: false}))
+
+    @supermodel.trackRequest(@courseInstance.fetch().then(=>
+      return if @destroyed
+      @teacherMode = @courseInstance.get('ownerID') is me.id
+
+      @owner = new User({_id: @courseInstance.get('ownerID')})
+      @supermodel.trackRequest(@owner.fetch())
+
+      classroomID = @courseInstance.get('classroomID')
+      @classroom = new Classroom({ _id: classroomID })
+      @supermodel.trackRequest(@classroom.fetch())
+
+      levelsLoaded = @supermodel.trackRequest(@levels.fetchForClassroomAndCourse(classroomID, @courseID, {
+        data: { project: 'concepts,type,slug,name,original,description' }
+      }))
+
+      @supermodel.trackRequest($.when(levelsLoaded, sessionsLoaded).then(=>
+        @buildSessionStats()
+        return if @destroyed
+        if @memberStats[me.id]?.totalLevelsCompleted >= @levels.size() - 1  # Don't need to complete arena
```

Comment:
Probably should be improved when new student views are implemented and behavior clarified. For now, this is moving around existing logic.